### PR TITLE
LastModifiedBy: BG and additions to common 

### PR DESCRIPTION
**Pull request by MoriaYechiel**

### CODE REVIEW

File: server/cmwell-bg/src/main/scala/cmwell/bg/Merger.scala [hunk 1]
```diff
@@ -72,6 +72,13 @@ class Merger(config: Config) extends LazyLogging {
     }
   }
 
+  private def merge_Modifiers (modifierA: String, modifierB: String): String = {
```
Comment (e-orz) [inline server/cmwell-bg/src/main/scala/cmwell/bg/Merger.scala]: use sorted set

Comment (e-orz) [reply server/cmwell-bg/src/main/scala/cmwell/bg/Merger.scala]: also, all merger should be refactdored

Comment (MoriaYechiel) [reply server/cmwell-bg/src/main/scala/cmwell/bg/Merger.scala]: Done

File: server/cmwell-bg/src/test/resources/indices_template.json [hunk 2]
```diff
@@ -160,6 +160,11 @@
                 }
               }
             },
+            "lastModifiedBy": {
+              "type": "keyword",
+              "store": true,
```
[the remark this answers is not in the document]
Comment (MoriaYechiel) [reply server/cmwell-bg/src/test/resources/indices_template.json]: Changed

File: server/cmwell-common/src/test/scala/cmwell/common/CommandSpec.scala [hunk 3]
```diff
@@ -70,10 +70,23 @@ class CommandSpec extends FlatSpec with Matchers with LazyLogging {
...
+    val currentTime = System.currentTimeMillis()
+    val overrideInfoton = new ObjectInfoton(
+      path = s"/cmt/cm/bg-test/re_process_ow/info_override",
+      dc = "dc",
+      indexTime = Some(currentTime),
+      lastModified = new DateTime(currentTime),
+      "Ori",
+      indexName = "cm_well_p0_0",
+      fields = Some(Map("a" -> Set(FieldValue("b"), FieldValue("c")))),
+      protocol = None
```
[the remark this answers is not in the document]
Comment (MoriaYechiel) [reply server/cmwell-common/src/test/scala/cmwell/common/CommandSpec.scala]: Yes

File: server/cmwell-common/src/test/scala/cmwell/common/CommandSpec.scala [hunk 4]
```diff
@@ -105,7 +118,15 @@ class CommandSpec extends FlatSpec with Matchers with LazyLogging {
           u_f.size should equal(cmdUpdate.updateFields.size)
           lm.getMillis should equal(cmdUpdate.lastModified.getMillis)
           lmb should equal(cmdUpdate.lastModifiedBy)
-        case OverwriteCommand(_, _) => ??? //TODO: add tests for OverwriteCommand
+        case OverwriteCommand(infoton, _) =>
```
[the remark this answers is not in the document]
Comment (MoriaYechiel) [reply server/cmwell-common/src/test/scala/cmwell/common/CommandSpec.scala]: Failed due to date conversions. (will show you)

File: server/cmwell-common/src/main/scala/cmwell/common/formats/JsonSerializer.scala [hunk 5]
```diff
@@ -346,6 +346,7 @@ object JsonSerializer extends AbstractJsonSerializer with LazyLogging {
     jsonGenerator.writeStringField("path", infoton.path)
     jsonGenerator.writeStringField("lastModified", dateFormatter.print(infoton.lastModified))
     jsonGenerator.writeStringField("lastModifiedBy", infoton.lastModifiedBy)
+    jsonGenerator.writeStringField("indexName", infoton.indexName)
```
[the remark this answers is not in the document]
Comment (MoriaYechiel) [reply server/cmwell-common/src/main/scala/cmwell/common/formats/JsonSerializer.scala]: Bug.. (Not mine)
I found it when I wrote the test to OverwriteCommand

File: server/cmwell-common/src/main/scala/cmwell/common/formats/JsonSerializer6.scala [hunk 6]
```diff
@@ -532,6 +532,15 @@ object JsonSerializer6 extends AbstractJsonSerializer with LazyLogging {
       s"expected value for 'lastModifiedBy' field\n${jsonParser.getCurrentLocation.toString}")
     val lastModifiedBy = jsonParser.getText()
 
+    assume(
```
[the remark this answers is not in the document]
Comment (MoriaYechiel) [reply server/cmwell-common/src/main/scala/cmwell/common/formats/JsonSerializer6.scala]: Same as up

File: server/cmwell-bg/src/test/scala/cmwell/bg/test/BGMergerSpec.scala [hunk 7]
```diff
@@ -135,14 +135,124 @@ class BGMergerSpec extends FlatSpec with Matchers with OptionValues {
...
+      now,
+      "Baruch2",
+      Map("last-name" -> Set(FieldValue("smith"))),
+      None
+    )
+    val change2 = ObjectInfoton(
+      "/bg-test-merge/objinfo2",
+      "dc1",
+      None,
+      now,
```
Comment (e-orz) [inline server/cmwell-bg/src/test/scala/cmwell/bg/test/BGMergerSpec.scala]: now+3 or something

File: server/cmwell-bg/src/main/scala/cmwell/bg/imp/CommandsSource.scala [hunk 8]
```diff
@@ -74,6 +74,12 @@ object CommandsSource extends LazyLogging {
             logger.error(s"deserialize command error for msg [$msg] and value: [$s]",err)
           }
           val command = commandTry.get
+
+          command match {
```
Comment (e-orz) [inline server/cmwell-bg/src/main/scala/cmwell/bg/imp/CommandsSource.scala]: add `case _ => nothing`

**Pull request merged**